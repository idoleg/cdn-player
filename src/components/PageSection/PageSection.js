import React from "react";
import classes from "./PageSection.module.css";

export default (props) => (
	<main className={[classes.PageSection, props.className].join(" ")}>
		<section className={classes.Content}>
			<header className={classes.Title}>{props.title || "Title"}</header>
			{props.children}
		</section>
	</main>
);
