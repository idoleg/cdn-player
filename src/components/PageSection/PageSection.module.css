.PageSection {
    display: flex;
    justify-content: center;
}

.Content {
    margin: 30px 20px;
    flex-grow: 1;
    max-width: 800px;
}

.Content>*:not(:last-child) {
    margin-bottom: 20px;
}

.Title {
    text-shadow: 0 1px 1px rgba(0,0,0, .15);
    font-weight: 700;
    font-size: 30px;
}