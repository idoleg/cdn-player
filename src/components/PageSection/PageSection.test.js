import React from "react";
import renderer from "react-test-renderer";
//import { mount } from "enzyme";
import PageSection from "./PageSection";

describe("PageSection", () => {

	it("render correctly empty PageSection component", () => {
		const PageSectionComponent = renderer.create(<PageSection />).toJSON();
		expect(PageSectionComponent).toMatchSnapshot();
  });

  it("render correctly empty PageSection component with className", () => {
		const PageSectionComponent = renderer.create(<PageSection className="test-class" />).toJSON();
		expect(PageSectionComponent).toMatchSnapshot();
	});

	it("render correctly PageSection component with content", () => {
		const PageSectionComponent = renderer.create(
			<PageSection header="It's my title">
				<div>It's my content</div>
			</PageSection>).toJSON();
		expect(PageSectionComponent).toMatchSnapshot();
	});

});
