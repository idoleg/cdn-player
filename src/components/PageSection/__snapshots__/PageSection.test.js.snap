// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSection render correctly PageSection component with content 1`] = `
<main
  className="PageSection"
>
  <section
    className="Content"
  >
    <header
      className="Title"
    >
      Title
    </header>
    <div>
      It's my content
    </div>
  </section>
</main>
`;

exports[`PageSection render correctly empty PageSection component 1`] = `
<main
  className="PageSection"
>
  <section
    className="Content"
  >
    <header
      className="Title"
    >
      Title
    </header>
  </section>
</main>
`;
